### DISCUSSION TITLE
How to get data from one web while deploying LLMs locally?

### DISCUSSION THREAD

**Discussion by xinyuliu006** [ggml-org/llama.cpp#8930] [Q&A]
**Expectation:** I want my project able to get data from _"purpleair.com"_ so that if I ask my model like "How is the air quality in London?" , it will search on that web for data and give me answer.
**Questions:** Is this idea feasible? What should I do to realize this?

**Comment by dspasyuk** [comment]
You can use RAG for it, so basically, get the user prompt, distill it using for example,  compromize.js, then use fetch to get the data you need and then embed that data into original user prompt as a context and feed it to the llama-cli.  
![example](https://github.com/user-attachments/assets/44a3513a-ce17-4533-9bb7-7504e523ad00)
